rcsb/utils/seq/UniProtUtils.py

```diff
@@ -412,16 +413,31 @@ def __doRequest(self, idList, retryAltApi=True, usePrimary=True):
 
     def __doRequestPrimary(self, idList):
         """ """
+        missingIds, demergedIdList = [], []
         baseUrl = self.__urlPrimary
         ureq = UrlRequestUtil()
         idListStr = "%2C%20".join(idList)
         endPoint = "uniprotkb/accessions?accessions=" + idListStr
         hD = {"Accept": "application/xml"}
+        logger.debug("Performing request with baseUrl, endpoint, and hD:  %r  %r  %r", baseUrl, endPoint, hD)
         ret, respCode = ureq.getUnWrapped(baseUrl, endPoint, paramD=None, headers=hD, overwriteUserAgent=False)
         if respCode != 200:
             logger.error("Primary request failed with retCode %r", respCode)
+            res = requests.get(os.path.join(baseUrl, endPoint))
+            while res.status_code == 400:
+                if "It should be a valid UniProtKB accession" in res.json()["messages"][0]:
```

Comment:
Why not loop through the messages and remove all the invalid IDs before sending the request again?

Reply:
I didn't do that because I haven't actually seen a case where there were more than one invalid IDs in a single response, so I'm not sure what format it to expect it in (e.g., multiple messages, or multiple IDs in the same message)...

Reply:
I tried adding extra characters in the IDs in the URL: [modified request](https://rest.uniprot.org/uniprotkb/accessions?accessions=xxxxxO66458%2C%20A6P7M9%2C%20Q9BSA4%2C%20P9WMM5%2C%20S7XVG3%2C%20Q609M8%2C%20Q86N96%2C%20A0A0H3JUY4%2C%20A0A1S3APZ1%2C%20P02185%2C%20A0A080IK26%2C%20P51687%2C%20Q1LBW6%2C%20P38123%2C%20Q92608%2C%20Q9X0H0%2C%20P14780%2C%20P41266%2C%20Q8U0M3%2C%20Q3HNQ2%2C%20W7XDM6%2C%20I7MK25%2C%20D3KYU3%2C%20A4HXQ3%2C%20Q7S4I4%2C%20P29600%2C%20A0A8J1MFV7%2C%20P62083%2C%20Q9P1F3%2C%20Q59WG0%2C%20Q5L3C7%2C%20P04303%2C%20P46683%2C%20Q5SKU4%2C%20A0A6H0HKY4%2C%20Q9Y821%2C%20A0A183CD52%2C%20P21866%2C%20Q77Q82%2C%20Q9SJE9%2C%20E1K248%2C%20Q6ZRY4%2C%20P56217%2C%20Q12018%2C%20C4M0U8%2C%20Q46WB6%2C%20Q8AZ32%2C%20P0CG30%2C%20P69771%2C%20Q9ICB7%2C%20A8I0K9%2C%20P93024%2C%20M4MD44%2C%20B4F320%2C%20B7XA03%2C%20Q9HK35%2C%20Q9UUT7%2C%20P07071%2C%20Q6CNW4%2C%20P33442%2C%20Q4Q712%2C%20CAA05093%2C%20G0CBC0%2C%20Q9EXR9%2C%20Q8Y8T4%2C%20B1K378%2C%20A0A0F6F6Q9%2C%20P11546%2C%20Q8G962%2C%20P0AB67%2C%20Q969H8%2C%20Q2FRL9%2C%20Q55460%2C%20Q2WBH2%2C%20Q81DD6%2C%20A0A0P6Y0P9%2C%20P00309%2C%20P25569%2C%20P18247%2C%20Q8KF02%2C%20Q8PMH0%2C%20Q9L8I7%2C%20Q15013%2C%20O82015%2C%20A1RAA6%2C%20P80340%2C%20O50335%2C%20Q6NCZ6%2C%20Q03347%2C%20Q97YD5%2C%20H6S0J1%2C%20A0A384E149%2C%20G1UIZ2%2C%20O15144%2C%20Q08204%2C%20A0A4V8GZZ2%2C%20O3xxxxxx1418%2C%20F1ARN3%2C%20P09622%2C%20A0A2G5I8W0)

Reply:
Ah, good idea. I'll update the code with that then, thanks.

Reply:
I wish they gave a representation of the errors that didn't require parsing this user-friendly text, but not much we can do about that. Hopefully they won't make any changes to the message.

Reply:
Yeah I agree, it's a mess of a string to parse and not ideal. Hopefully they don't change it too soon...